bug: fails to build with LLVM libc++

### Summary

WasmEdge fails to build with LLVM LibC++.

### Current State

Fails to compile due to specific warnings being marked as errors by clang. 

### Expected State

Successful compilation

### Reproduction steps

Build wasmedge with llvm libc++


### Screenshots

_No response_

### Any logs you want to share for showing the specific issue

```
home/theo/src/WasmEdge/include/common/int128.h:550:20: error: 'is_class' cannot be specialized: Users are not allowed to specialize this standard library entity [-Winvalid-specialization]
  550 | template <> struct is_class<WasmEdge::uint128> : std::true_type {};
      |                    ^
/home/theo/src/llvm-project/build/bin/../include/c++/v1/__type_traits/is_class.h:22:29: note: marked 'no_specializations' here
   22 | struct _LIBCPP_TEMPLATE_VIS _LIBCPP_NO_SPECIALIZATIONS is_class : public integral_constant<bool, __is_class(_Tp)> {};
      |                             ^
/home/theo/src/llvm-project/build/bin/../include/c++/v1/__config:1186:9: note: expanded from macro '_LIBCPP_NO_SPECIALIZATIONS'
 1186 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^

```

### Components

C SDK

### WasmEdge Version or Commit you used

9baa2937e281e6afcf270739aa79d84da0c486f4

### Operating system information

Fedora Asahi Remix

### Hardware Architecture

aarch64

### Compiler flags and options

CMake version: 4.0.0
I am using a custom LLVM trunk build with LLD and llvm runtimes (compiler-rt, libcxx, libunwind). I am still using the system (g)libc from Fedora though.

Your build runner only covers 14 and 15,
```
      matrix: "[{'name':'MacOS 15 (x86_64)','runner':'macos-15-intel','darwin_version':24,'arch':'x86_64'},
                {'name':'MacOS 14 (arm64)','runner':'macos-14','darwin_version':23,'arch':'arm64'}]"
```
and the error only happen with macos-26 runner. 

```
  In file included from /private/tmp/wasmedge-20260415-7685-s8dzni/wasmedge/include/common/endian.h:17:
  /private/tmp/wasmedge-20260415-7685-s8dzni/wasmedge/include/common/int128.h:545:20: error: 'is_class' cannot be specialized: Users are not allowed to specialize this standard library entity [-Winvalid-specialization]
    545 | template <> struct is_class<WasmEdge::uint128> : std::true_type {};
        |                    ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX26.sdk/usr/include/c++/v1/__type_traits/is_class.h:22:8: note: marked '_Clang::no_specializations' here
     22 | struct _LIBCPP_NO_SPECIALIZATIONS is_class : integral_constant<bool, __is_class(_Tp)> {};
        |        ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX26.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '_LIBCPP_NO_SPECIALIZATIONS'
   1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
        |         ^
```

MacOSX26.sdk may limit some feature, not sure.

--- a/include/common/int128.h
+++ b/include/common/int128.h
@@ -540,11 +540,6 @@ public:
 };
 } // namespace std
 
-#include <type_traits>
-namespace std {
-template <> struct is_class<WasmEdge::uint128> : std::true_type {};
-} // namespace std
-
 namespace WasmEdge {
 // If there is a built-in type __int128, then use it directly
 #if defined(__x86_64__) || defined(__aarch64__) ||                             \

